### PR TITLE
Bly 2161

### PR DESCRIPTION
**Pull request by FiboApe**

### CODE REVIEW

File: apps/root/src/common/components/wallet-selector/index.tsx [hunk 1]
```diff
@@ -139,7 +206,7 @@ const WalletSelector = ({ options, size = 'small' }: WalletSelectorProps) => {
       }
     };
 
-    if (prevWallets && prevWallets.length > 0 && wallets.length > prevWallets.length) {
+    if (prevWallets && prevWallets.length > 0 && wallets.length !== prevWallets.length) {
```
Comment (th0rOdinson) [inline apps/root/src/common/components/wallet-selector/index.tsx:209]: unlink wallet is in the house!

File: packages/ui-library/src/components/options-menu/index.tsx [hunk 2]
```diff
@@ -39,14 +39,132 @@ type MenuOption = {
...
+
+type OptionsMenuOption = DividerOption | MenuOptionWithOptions;
 
 interface OptionsMenuItemsProps {
   options: OptionsMenuOption[];
   handleClose: () => void;
   anchorEl: HTMLElement | null;
 }
 
+const DividerItem = () => <Divider />;
```
Comment (th0rOdinson) [inline packages/ui-library/src/components/options-menu/index.tsx:56]: What diff does this make?

File: apps/root/src/common/components/edit-label-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        {
+          label: <FormattedMessage description="renameWallet Save" defaultMessage="Save" />,
+          color: 'primary',
+          variant: 'contained',
+          onClick: handleEditLabel,
+          disabled: walletLabel === '',
+        },
+      ]}
+      actionsAlignment="horizontal"
+    >
```
[the remark this answers is not in the document]
Comment (FiboApe) [reply apps/root/src/common/components/edit-label-modal/index.tsx:88]: Oh right forgot the self-close modal. For the loading indicator I would actually just close it and update the value, if the BE fails, we revert back

File: apps/root/src/common/components/unlink-wallet-modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+const StyledContactItem = styled(ContainerBox).attrs(() => ({ justifyContent: 'space-between', alignItems: 'center' }))`
+  ${({ theme: { palette, spacing } }) => `
+  border: 1px solid ${colors[palette.mode].border.border1};
+  border-radius: ${spacing(2)};
+  background-color: ${colors[palette.mode].background.secondary};
+  transition: background 200ms ease-in-out;
+  padding: ${spacing(2.25)} ${spacing(4)};
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
[the remark this answers is not in the document]
Comment (FiboApe) [reply apps/root/src/common/components/unlink-wallet-modal/index.tsx:17]: good catch

File: apps/root/src/common/components/edit-label-modal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+      onClose={onCancel}
+      maxWidth="sm"
+      title={<FormattedMessage description="renameWallet title" defaultMessage="Rename your wallet" />}
+      actions={[
+        {
+          label: <FormattedMessage description="renameWallet Save" defaultMessage="Save" />,
+          color: 'primary',
+          variant: 'contained',
+          onClick: handleEditLabel,
+          disabled: walletLabel === '',
```
[the remark this answers is not in the document]
Comment (FiboApe) [reply apps/root/src/common/components/edit-label-modal/index.tsx:84]: Sure

File: packages/ui-library/src/components/options-menu/index.tsx (hunk 2 above)
Comment (FiboApe) [reply packages/ui-library/src/components/options-menu/index.tsx:56]: None, just thought it became a bit more legible to have a component type for each type of OptionMenuType, wdyt?

Comment (th0rOdinson) [reply packages/ui-library/src/components/options-menu/index.tsx:56]: Oh i get it. Looks good!

**Review by th0rOdinson — APPROVED** [review]

**Pull request merged**